src/dl.cpp

```diff
@@ -68,7 +70,25 @@ dl_error_t dl_instance_load( dl_ctx_t             dl_ctx,          dl_typeid_t
 	DL_ASSERT( (uint8_t*)instance + header->instance_size > packed_instance || (uint8_t*)instance < packed_instance + header->instance_size );
 	memcpy( instance, packed_instance + sizeof(dl_data_header), header->instance_size );
 
-	dl_internal_patch_instance( dl_ctx, root_type, (uint8_t*)instance, 0x0, (uintptr_t)instance );
+	if( header->version == 1 )
+		dl_internal_patch_instance( dl_ctx, root_type, (uint8_t*)instance, 0x0, (uintptr_t)instance );
+	else if( header->need_slow_patching )
```

Comment:
is there a reason we need to store this, can't this be calculated from the instance-size? I could guess if you are on 32-bit and can only address x bytes but you pointers only reside under that you would miss out on some optimizations... but how often will that happen?

Reply:
and maybe a name such as 'uses_ptr_chain_patching' or similar that better describes the method would be a better name?

Reply:
I made the "fast pointer patch" optional, so there are two code paths which always store pure offsets, and never link to the next offset. So yes, right now we need to know how the offset are meant to be decoded.

Reply:
And feel free to rename, I am not picky about the wording.